### ISSUE TITLE
Collapsible legend

### ISSUE THREAD

**Issue by joshm-3b6b58** [labrad/scalabrad-web#229]
For multiple 1D traces it would be great to have a lengend to know which trace is which. It would be also great if it were collapsable to it didn't always take up valuable real-estate on the screen. Having it enabled through the URL would be nice too.

**Comment by adamcw** [comment]
I agree, the line graphs are frustrating without a legend in my opinion.

A HTML implementation would be trivial, but if we're aiming to ever resolve issue #184 then it would have to be implemented in WebGL.

**Comment by joshm-3b6b58** [comment]
Yes good point. So integrating WebGL and DOM elements into the same canvas or something is basically impossible?

**Comment by adamcw** [comment]
I am currently investigating it as a solution to the Zoom Box issues in the WebGL branch. I don't think we can do DOM, but inserting SVG might be possible.

**Comment by adamcw** [comment]
Just an update given the previous comments. It looks like if there is a short-term screenshot solution it will render SVG and WebGL into a single canvas and then screenshot that. That means any implementation of a feature that we desire to screenshot should be either WebGL or SVG.

HTML/DOM elements can technically be rendered into SVG and then merged, but I think this is already a fairly complex mash of technologies, I'd like to avoid it spiraling further by going down that path.

**Comment by zchen088** [comment]
+1 Not having a legend makes interpreting the 1D traces really difficult.
